shims/spark300/src/main/scala/com/nvidia/spark/rapids/shims/spark300/Spark300Shims.scala

```diff
@@ -415,4 +415,17 @@ class Spark300Shims extends SparkShims {
       resolver: Resolver): Unit = {
     GpuSchemaUtils.checkColumnNameDuplication(schema, colType, resolver)
   }
+
+  override def sortOrderChildren(s: SortOrder): Seq[Expression] = {
+    (s.sameOrderExpressions + s.child).toSeq
+  }
+
+  override def sortOrder(
+      child: Expression,
+      direction: SortDirection,
+      nullOrdering: NullOrdering): SortOrder = SortOrder(child, direction, nullOrdering, Set.empty)
+
+  override def copySortOrderWithNewChild(s: SortOrder, child: Expression): SortOrder = {
```

Comment:
I might have missed how this implementation isn't identical to that in `Spark310Shims`. If it is indeed identical, why did this copy functionality need to be shimmed?

Reply:
It's identical source code but compiles differently.  Source vs. binary compatibility.